WIP: Added support for inline scripting

wattry opened the pull request:
## Description
Adding inline/Adhoc script support would allow devs to included much needs functionality seeing as no package manager is included, I believe this would enhance user experience at very little cost while allowing users to write short automation scripts.

I discovered this package while reading [this article dev](https://dev.to/vonheikemen/a-simple-way-to-replace-npm-scripts-in-deno-4j0g). I liked the package however it was not quite working how I needed it, nor how the OP was discussing.

**Types of changes**
- New feature (non-breaking change which adds functionality)

**Related links**
https://dev.to/vonheikemen/a-simple-way-to-replace-npm-scripts-in-deno-4j0g

## Changelog
#### Added
* replaceEnvVars + respective types and tests
* _runInlineScript + respective types and tests
* import dotenv

wattry committed:
Added support for inline scripting

wattry committed:
Fixed linting issues.

BentoumiTech committed:
Merge branch 'master' into wattry/issue-14

BentoumiTech commented on src/utils/replaceEnvVars.ts:
```diff
@@ -0,0 +1,28 @@
+  }
+
+  // I'm not sure if you have a loop preference? This is the method I have become accustom to in terms of readability and performance.
+  envKeys.forEach((envKey: string) => {
```
The `forEach` looks good for me

BentoumiTech commented on src/utils/replaceEnvVars.ts:
```diff
@@ -0,0 +1,28 @@
+    // This is a little ugly but I like explicit variable declaration for readability.
+    const envValue: string = env[envKey];
+    // String are immutable so it doesn't matter if we directly work on the cmd string.
+    cmd = cmd?.replace(`\${${envKey}}`, envValue);
```
What do you think about changing the format from `${ENV_NAME}` to `$ENV_NAME`?

I prefer the second as it's the way POSIX commands utilize environment variables, it's the same way you would write in a bash terminal for example.

wattry replied:
I am completely fine with changing that

rivy replied:
FYI, in POSIX shells `${ENV_NAME}` == `$ENV_NAME`.

While `$ENV_NAME` is commonly used, `${ENV_NAME}` is less ambiguous when embedded within other text (eg, "${ENV_NAME}S" vs "$ENV_NAMES"). Given the possible ambiguities, you'll want to at least allow the `${ENV_NAME}` format.

BentoumiTech commented:
@wattry Thank you for the contribution I quickly went through it and added a couple of comments. 

I'm currently finishing environment variable support #25 I think this could be helpful for this PR.

Anything I could do to help with the PR?

wattry commented:
> @wattry Thank you for the contribution I quickly went through it and added a couple of comments.
> 
> I'm currently finishing environment variable support #25 I think this could be helpful for this PR.
> 
> Anything I could do to help with the PR?

In my PR I implemented basic env support in my code so I would have to just remove that and wait on your env support to be released. As of now I don't think there is anything but if you have any suggestions I'm open to them

BentoumiTech commented:
I think it's okay, once I finish the `env` PR I'll merge it in your PR then release.
